python/tidy/tidy.py

```diff
@@ -376,6 +381,38 @@ def check_ruff_lints():
             )
 
 
+@dataclass
+class PyreflyDiagnostic:
+    """
+    Represents a single diagnostic error reported by Pyrefly.
+    """
+
+    line: int
+    column: int
+    stop_line: int
+    stop_column: int
+    path: str
+    code: int
+    name: str
+    description: str
+    concise_description: str
+
+
+def run_python_type_checker() -> Iterator[Tuple[str, int, str]]:
+    print("\r ➤  Running `pyrefly` checks...")
```

Comment:
Maybe this can also be more generic:

```suggestion
    print("\r ➤  Checking type annotations in python files ...")
```